expression/builtin_cast.go

```diff
@@ -446,6 +448,24 @@ func (b *builtinCastIntAsIntSig) evalInt(row chunk.Row) (res int64, isNull bool,
 	return
 }
 
+func (b *builtinCastIntAsIntSig) supportReverseEval() bool {
+	return true
+}
+
+func (b *builtinCastIntAsIntSig) reverseEval(sc *stmtctx.StatementContext, res types.Datum, rType types.RoundingType) (val types.Datum, err error) {
+	resVal := res.GetInt64()
+	switch x := b.args[0].(type) {
+	case *Column:
+		return types.NewIntDatum(resVal), nil
+	case *Constant:
+		return types.Datum{}, errors.Errorf("invalid args for reverse evaluation, " +
+			"the expression should have exactly one column")
```

Comment:
Why do we return an error here instead of `types.NewIntDatum(resVal), nil`?

Reply:
Since the `arg[0]` for a `Cast` function is a constant. That means there is none column for the function. That's not the necessary condition for reverse evaluation.